### PR TITLE
gitlab: 10.8.0 -> 11.3.0, gitlab-ee: init at 11.3.0

### PR DESCRIPTION
**Pull request by krav**
gitlab-shell: 7.1.2 -> 8.1.1, nixos/gitlab: fixes

###### Motivation for this change

This adds support for GitLab enterprise edition.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by globin — DISMISSED** [review]
Small question and per GITLAB_WORKHORSE_VERSION and GITALY_SERVER_VERSION in the gitlab repository, workhorse should be updated to 5.1.0 and gitaly to 0.117.2

Thank you otherwise! I was planning on doing an update the next few days but you saved me some time!

File: pkgs/applications/version-management/gitlab-shell/default.nix [hunk 1]
```diff
@@ -1,20 +1,17 @@
-{ stdenv, ruby, bundler, fetchFromGitLab, go }:
+{ stdenv, ruby, bundler, fetchgit, go }:
 
 stdenv.mkDerivation rec {
-  version = "7.1.2";
+  version = "8.1.1";
   name = "gitlab-shell-${version}";
 
-  srcs = fetchFromGitLab {
```
[the remark this answers is not in the document]
Comment (krav) [reply pkgs/applications/version-management/gitlab-shell/default.nix]: Just a brain fart. I somehow managed to read it as fetchFromGithub, which has a stale mirror. I've changed it back, and upgraded workhorse and gitaly.

**Comment by andir** [comment]
Could you update this to the latest release? 11.2.3 was released today.

Should this also be backported?

**Comment by krav** [comment]
@globin Done. This should probably be backported as there is a security release 10.8.6 out and current in 18.03 is 10.8.0.

**Comment by afrepues** [comment]
Can you split your commits to make them atomic? That would make them easier to review and rollback if necessary. This would mean having separate commits for:

- the introduction of the enterprise version
- the different updates in versions
- changes to the gitlab module
- other fixes/changes

**Comment by krav** [comment]
@afrepues Done

**Comment by globin** [comment]
@GrahamcOfBorg build gitlab, gitlab-workhorse, gitlab-shell, gitaly

**Comment by globin** [comment]
@GrahamcOfBorg build gitlab gitlab-workhorse gitlab-shell gitaly

**Pull request merged**